lib/AST/Decl.cpp

```diff
@@ -2090,6 +2090,17 @@ bool swift::conflicting(ASTContext &ctx,
   if (!conflicting(sig1, sig2, skipProtocolExtensionCheck))
     return false;
 
+  // Two enum elements always conflict with each other. At this point, they
+  // have the same base name but different types.
+  if (sig1.IsEnumElement == sig2.IsEnumElement) {
+    return true;
+  }
+
+  // A function does not conflict with an enum element if their types differ.
```

Comment:
What if sig1 is an enum element and sig2 is a function? Enum cases can appear anywhere inside an enum body, including after functions.

Reply:
@slavapestov True, I have added that case now.